activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -614,6 +625,11 @@ def has_default_function?(default_value, default)
         def load_additional_types(oids = nil)
           initializer = OID::TypeMapInitializer.new(type_map)
 
+          if additional_type_records_cache.present? && oids.blank?
```

Comment:
Can we have a cache per oids if additional_type_records_cache is a Hash?  Seems like an easy win to cache those as well.

Reply:
The `oids` is an array that may not properly be a hash key.

Reply:
I saw your point, the `oids` may have been cached already. PR updated accordingly 👍 .